auth/server/src/main/java/com/seagates3/authorization/PublicAccessAuthorizer.java

```diff
@@ -0,0 +1,45 @@
+package com.seagates3.authorization;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public
+class PublicAccessAuthorizer {
+ private
+  static final String RESTRICTED_PUBLIC_ACCESS_ACTIONS =
+      "/RestrictedPublicAccessS3Actions.json";
+ private
+  static Set<String> actionsSet = new HashSet<>();
+ public
+  static void init() {
+    try {
+      InputStream in = PublicAccessAuthorizer.class.getResourceAsStream(
+          RESTRICTED_PUBLIC_ACCESS_ACTIONS);
+      InputStreamReader reader = new InputStreamReader(in, "UTF-8");
+      JsonParser jsonParser = new JsonParser();
+      JsonObject element = (JsonObject)jsonParser.parse(reader);
+      for (JsonElement action : element.get("actions").getAsJsonArray()) {
+        actionsSet.add(action.getAsString());
+      }
+    }
+    catch (IOException e) {
+    }
```

Comment:
We need to log the exception and throw the exception above.

Reply:
fixed